extensions/unknown/vectors.js

```diff
@@ -0,0 +1,180 @@
+            ),
+            arguments: {
+              ID: { type: Scratch.ArgumentType.STRING, defaultValue: "vec1" },
+              DIRECTION: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 90,
+              },
+              MAGNITUDE: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 10,
+              },
+            },
+          },
+          {
+            opcode: "vectorDelete",
+            blockType: Scratch.BlockType.COMMAND,
+            text: Scratch.translate("delete vector with [ID]"),
+            arguments: {
+              ID: { type: Scratch.ArgumentType.STRING, defaultValue: "vec1" },
+            },
+          },
+          {
+            opcode: "vectorChangedir",
+            blockType: Scratch.BlockType.COMMAND,
+            text: Scratch.translate(
+              "change vector [ID]'s direction to [DIRECTION]"
+            ),
+            arguments: {
+              ID: { type: Scratch.ArgumentType.STRING, defaultValue: "vec1" },
+              DIRECTION: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 90,
+              },
+            },
+          },
+          {
+            opcode: "vectorMag",
+            blockType: Scratch.BlockType.REPORTER,
+            text: Scratch.translate("magnitude of vector [ID]"),
+            arguments: {
+              ID: { type: Scratch.ArgumentType.STRING, defaultValue: "vec1" },
+            },
+          },
+          {
+            opcode: "listOfIDs",
+            blockType: Scratch.BlockType.REPORTER,
+            text: Scratch.translate("List of IDs"),
+          },
+          {
+            opcode: "vectorDir",
+            blockType: Scratch.BlockType.REPORTER,
+            text: Scratch.translate("direction of vector [ID]"),
+            arguments: {
+              ID: { type: Scratch.ArgumentType.STRING, defaultValue: "vec1" },
+            },
+          },
+        ],
+      };
+    }
+
+    vectorCreate(args) {
+      const id = Scratch.Cast.toString(args.ID);
+      if (!ids.includes(id)) ids.push(id);
+      directions[id] = Scratch.Cast.toNumber(args.DIRECTION);
+      magnitudes[id] = Scratch.Cast.toNumber(args.MAGNITUDE);
+    }
+
+    vectorChange(args) {
+      const id = Scratch.Cast.toString(args.ID);
+      directions[id] = Scratch.Cast.toNumber(args.DIRECTION);
+      magnitudes[id] = Scratch.Cast.toNumber(args.MAGNITUDE);
+    }
+
+    vectorChangedir(args) {
+      const id = Scratch.Cast.toString(args.ID);
+      directions[id] = Scratch.Cast.toNumber(args.DIRECTION);
+    }
+
+    vectorChangeMag(args) {
+      const id = Scratch.Cast.toString(args.ID);
+      magnitudes[id] = Scratch.Cast.toNumber(args.MAGNITUDE);
+    }
+
+    vectorMag(args) {
+      const id = Scratch.Cast.toString(args.ID);
+      return magnitudes[id] !== undefined
+        ? magnitudes[id]
+        : Scratch.translate("null");
```

Comment:
translating block return valeus is usually not a good idea; people may compare the result of this block to the string `null` (or whatever it is in languages) and then have it break in other languages that have a different string for it.
(`null` specifically is a fairly language-neutral word so it should be fine untranslated. or follow the Scratch precedent and return an empty string; this is what lists also do for example so I don't think it would be confusing)